reference/configuration/framework.rst

```diff
@@ -545,138 +1214,66 @@ If this option is enabled, validation constraints can be defined using annotatio
-                name:                 ~
-                cookie_lifetime:      ~
-                cookie_path:          ~
-                cookie_domain:        ~
-                cookie_secure:        ~
-                cookie_httponly:      ~
-                gc_divisor:           ~
-                gc_probability:       ~
-                gc_maxlifetime:       ~
-                save_path:            "%kernel.cache_dir%/sessions"
-
-            # serializer configuration
-            serializer:
-               enabled: false
-
-            # templating configuration
-            templating:
-                assets_version:       ~
-                assets_version_format:  "%%s?%%s"
-                hinclude_default_template:  ~
-                form:
-                    resources:
+**type**: ``string`` **default**: ``'file'``
 
-                        # Default:
-                        - FrameworkBundle:Form
-                assets_base_urls:
-                    http:                 []
-                    ssl:                  []
-                cache:                ~
-                engines:              # Required
-
-                    # Example:
-                    - twig
-                loaders:              []
-                packages:
+This option can be one of the following values:
+
+file
+    Use the filesystem to cache annotations
+none
+    Disable the caching of annotations
+a service id
+    A service id referencing a `Doctrine Cache`_ implementation
+
+file_cache_dir
+..............
+
+**type**: ``string`` **default**: ``'%kernel.cache_dir%/annotations'``
+
+The directory to store cache files for annotations, in case
+``annotations.cache`` is set to ``'file'``.
+
+debug
+.....
 
-                    # Prototype
-                    name:
-                        version:              ~
-                        version_format:       "%%s?%%s"
-                        base_urls:
-                            http:                 []
-                            ssl:                  []
-
-            # translator configuration
-            translator:
-                enabled:              false
-                fallback:             en
-
-            # validation configuration
-            validation:
-                enabled:              false
-                cache:                ~
-                enable_annotations:   false
-                translation_domain:   validators
-
-            # annotation configuration
-            annotations:
-                cache:                file
-                file_cache_dir:       "%kernel.cache_dir%/annotations"
-                debug:                "%kernel.debug%"
+**type**: ``boolean`` **default**: ``%kernel.debug%``
+
+Whether to enable debug mode for caching. If enabled, the cache will
+automatically update when the original file is changed (both with code and
+annotation changes). For performance reasons, it is recommended to disable
+debug mode in production.
```

Comment:
... to disable debug mode in production, which will happen automatically if you use the default value (i.e. ``%kernel.debug%).